R/forms.R

```diff
@@ -390,20 +435,46 @@ createFormSchemaFromData <- function(x, databaseId, label, folderId = databaseId
       is.character(codes)&&
       length(codes)==ncol(x)&&
       length(unique(codes[!is.na(codes)]))==length(codes[!is.na(codes)])
+  )
+  if (!is.null(parentFormId)) {
+    stopifnot("The parentFormId must be a character string if defined" = (is.character(parentFormId)&&length(parentFormId)==1))
+    stopifnot("The parentIdColumn must be a character string if defined" = 
+                is.null(parentIdColumn)||
+                (is.character(parentIdColumn)&&length(parentIdColumn)==1)
     )
-
+  } else {
+    if (!is.null(parentIdColumn)) {
+      warning("parentIdColumn defined without a parentFormId. Ignoring parentIdColumn and including the column in the schema.")
+      parentIdColumn = NULL
+    }
+  }
+  
   providedCols <- names(x)
-
+  
+  if (!is.null(parentFormId)&&!is.null(parentIdColumn)) {
+    providedCols[providedCols != parentIdColumn]
+  }
+  
   stopifnot("Some key columns do not exist in the data.frame provided" = keyColumns %in% providedCols)
   stopifnot("Some required columns do not exist in the data.frame provided" = keyColumns %in% providedCols)
-
-  fmSchema <- formSchema(databaseId = databaseId, label = label, folderId = folderId)
+  stopifnot("The parentIdColumn does not exist in the data.frame provided" = parentIdColumn %in% providedCols)
+  
+  if(!missing(folderId)) {
+    if (!missing(parentId) && !is.null(parentId) && folderId != parentId) {
+      warning("folderId does not match parentId. folderId is deprecated in createFormSchemaFromData(). Ignoring folderId and using parentId.")
+      folderId = parentId
```

Comment:
This is odd, we should only be doing this when parentId is NULL and folderId is provided

Reply:
This needs to be the same logic as formSchema

Reply:
This is definitely a bug as folderId is now never provided when uploading schema